bindings/js/api_test.js

```diff
@@ -54,16 +54,9 @@ var passTest = async function(func_sig, func, msg, t) {
   });
   var result = await func.apply(null, args);
   if (func_sig.return.type !== "void") {
-    var result_val;
```

Comment:
@denismerigoux can you check this?

Reply:
So here @victor-dumitrescu you removed a small optimization I had made. Indeed, normally all API functions return an array containing the tuple of the results. But in the case where the tuple had one element, I was removing the array and directly returning the element. Do you remember why you changed this ?

Reply:
There were a few things happening here. All functions in this category return a tuple of (result, output buffers) but we only care about the result at this stage, so it's fine to discard the rest. Here this was done if the function has any output buffers specified in api.json. However, even if that's not the case the function will return a tuple of (result, []) so I eliminated this check and I'm always taking the first part of the tuple as the result. Otherwise in this case result_val would be something like `[ true, [] ]`

Then, the result_val / expected_result check was broken, it should have been an inequality since if it's true we throw an error. The reason this wasn't being triggered is that this was comparing strings (as returned by `postprocessing`) with bool/ints (as returned by the function), which would always be false with === and so the test would pass. I added the posprocessing for the return value as well and fixed the check.

Reply:
Oh thank you, I had not spotted this. Damn Javascript operators... I approve then :)